app/views/sufia/homepage/_recent_items.html.erb

```diff
@@ -0,0 +1,30 @@
+<%
+#First, put a bunch of eligible works into a bag:
+conditions =  {'read_access_group_ssim'=>'public'}
+how_many_works_in_bag = 15
+sort_by = ["system_modified_dtsi desc"]
+opts = {:rows=>how_many_works_in_bag, :sort=>sort_by}
+works_to_pick_from = GenericWork.search_with_conditions( conditions, opts)
+
+# Now, pick a few of these out of the bag at random to show.
+# Shuffle the bag again every few minutes.
+how_often_to_change=60 #seconds
+srand Time.now.to_i/how_often_to_change
+how_many_works_to_show = 6
+works_to_show = works_to_pick_from.sort_by{rand}[0..how_many_works_to_show-1]
+%>
+
```

Comment:
this should not be inline in the view. It should be in the controller or a helper method.

Reply:
I am not sure what your "how often to change" implementation is doing, or if it's working.   What do you think it's doing?

Reply:
if we're going to be doing the fetch every time anyway, might as well just have it change on every page load. The only point I see to having it say 'stable' for eg 60 seconds is to avoid the performance hit of fetching em on every view. Is it bad UX if it changes on every page load?

Reply:
Will move to the controller, of course! Just wanted to have everything in one place as I mess with the design a bit; I'm already changing thing around.
I do like having things change a bit more slowly than on every page view -- ideally I'd like it to change every 10 minutes or so but I wanted to make sure it was working and it's easier to test if tit changes every minute.
The use case for having things change a bit more slowly is: "I found something cool serendipitously on the home page, and I want it to be there in a couple more minutes once I switch e.g. from my phone to my desktop."